src/Simulation/qdk_sim_rs/src/linalg/array_ext.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Private extension traits and methods for working with arrays.
+
+use ndarray::{
+    Array, ArrayBase, Axis, Data, DataMut, Dimension, Ix2, OwnedRepr, RawData, RemoveAxis,
+};
+use num_traits::Zero;
+
+use crate::error::QdkSimError;
+
+pub(crate) trait ShapeExt {
+    fn require_square(&self) -> Result<usize, QdkSimError>;
+}
+
+impl<S, D> ShapeExt for ArrayBase<S, D>
+where
+    S: RawData,
+    D: Dimension,
```

Comment:
So for extension traits there's [RFC 445](https://rust-lang.github.io/rfcs/0445-extension-trait-conventions.html). It looks like the traits in this module don't follow those naming conventions.

Reply:
Ah, that it should be `ArrayBaseExt`?

Reply:
Yeah, I think so.

- `ShapeExt` -> `ArrayBaseExt`
- `MatrixExt` -> `ArrayBaseMatrixExt`
- `RemoveAxisExt` -> `ArrayBaseRemoveAxisExt`

(Of course, assuming that there will never be any other impls for these - for example, n-d arrays represented by something other than an ArrayBase. Extension traits are kind of weird since the line between a single-impl trait and a many-impl trait seems blurry.)

Reply:
Sounds good, I'll go on and modify, then.